Similar to commit 67afddbf431140c1ee064bf79a2fa5a95575488e with title
"environment: Add LLVM 18 support to get_llvm_tool_names()", but with
minor version following the upstream versioning change.

--- a/mesonbuild/environment.py
+++ b/mesonbuild/environment.py
@@ -188,6 +188,7 @@ def get_llvm_tool_names(tool: str) -> T.List[str]:
     # unless it becomes a stable release.
     suffixes = [
         '', # base (no suffix)
+        '-18.1',  '181',
         '-17',  '17',
         '-16',  '16',
         '-15',  '15',
